ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,32 +31,216 @@ public  void Init()
+            m_Game = new Game(BoardSize, GameType);
+           // while(true)
+            //{
+            printGameBoard();
+            playTurn();
+            ReadLine();
+            
+            //}
+        }
+
+        private void playTurn()
+        {
+            int row = 0;
+            int col = 0;
+            //int return status: continue, place_taken, finished game
+            
+            if(m_Game.CurrentTurn == 1 || m_Game.GameType == Game.eGameTypes.PVP)
+            {
+Label: 
+                getRowColFromUser(out row, out col);
+                string msg ="";
+                int retStatus;
+                if (row == QUIT) // if press Q
+                {
+                    m_Game.Playerforfeit();
+                     msg = getEndGameMsg(Game.StatusWin);
+                }
+                // if Place is taken/ win/ tie
+                else if(!m_Game.PlaceSymbolAndReturnStatus(row, col, out retStatus))
+                {
+                    if(retStatus == Game.StatusTaken)
+                    {
+                        WriteLine("Place is taken!");
+                        goto Label;
+                        
+                    }
+                    else
+                    {
+                         msg = getEndGameMsg(retStatus);
+                    }
+                }
+                WriteLine(msg);
+            }
+            else
             {
 
             }
         }
 
-        private void getBoardSizeFromUser()
+        private string getEndGameMsg(int i_RetStatus)
         {
-            string userInputStr;
+            StringBuilder stringBuilder = new StringBuilder();
+            if(i_RetStatus == Game.StatusWin)
+            {
+                //write who wins
+            }
+            else
+            {
+                stringBuilder.Append("This is a Tie!");
+            }
+           // stringBuilder.AppendFormat("Score is: {0} Player1: {1}", Environment.NewLine, m_Game
+            return stringBuilder.ToString();
+        }
+
+        private void getRowColFromUser(out int i_Row, out int i_Col)
+        {
+            string turnMsg = GetTurnMessage();
+            string userTurnInput;
+            do
+            {
+                WriteLine(turnMsg);
+                userTurnInput = ReadLine();
+            }
+            while (!turnInputIsValid(userTurnInput, out i_Row, out i_Col));
+        }
+
+        private bool turnInputIsValid(string i_UserTurnInput, out int o_Row, out int o_Col)
+        {
+            o_Row = QUIT;
+            o_Col = QUIT;
+            const bool v_IsValid = true;
+            bool turnInputIsValid = !v_IsValid;
+            i_UserTurnInput = removeWhitSpaces(i_UserTurnInput);
+            if (i_UserTurnInput == QUITSYMBOL)
+            {
+                turnInputIsValid = v_IsValid;
+            }
+            // Expecting here to have the following string: 
+            // "i,j" - where i is row and j is col.
```

Comment:
This input check works specifically when the board is smaller that 10X10. but it will not work for inputs such as "21, 16", "321321,234234" and so on, i think it's better to make it more reusable

and also (not really an issue, just a thought) , forcing the user to the speciefic format "x,y" seems kinda strict, maybe we can be more flexible? like allowing white spaces, not forcing "," in the middle and so on.